### PR TITLE
Adding security tests for User Management

### LINKED ISSUE #9142: User Management views are missing security testing
### Issue Summary
Views in the User Management component are not tested against unauthorized users.

### Steps to Reproduce

1. Remove all authorization decorators in `wagtail.users.views.users`.
2. Run `./runtests.py`

All the tests will pass as a result of executing the test runner.

### PR DESCRIPTION
**Pull request by mehrdadmoradii**
fixes #9142

Writing tests to make sure only authorized users can access the resources in `wagtail.users.views.users.py`.

### CODE REVIEW

**Comment by lb-** [comment]
Wow. What a legend @mehrdadmoradii - thank you.

**Comment by mehrdadmoradii** [comment]
@lb- thanks for the review!

File: wagtail/users/tests/test_admin_views.py [hunk 1]
```diff
@@ -1806,3 +1806,257 @@ def test_get_group_viewset_cls_custom_form_does_not_exist(self):
                 'Module "wagtail.users.tests" does not define a "CustomClassDoesNotExist" attribute/class',
                 str(exc_info.exception),
             )
+
+
+class TestAuthorizationIndexView(TestCase, WagtailTestUtils):
```
Comment (lb-) [inline wagtail/users/tests/test_admin_views.py:1811]:
```suggestion
class TestAuthorisationIndexView(TestCase, WagtailTestUtils):
```

Small nitpick but we do try to use English UK (not US) spelling when in the code or documentation.

Note, if referring to something like [`HTTP Authorization`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization) we would use US spelling but a variable or test case should use the UK spelling.

https://docs.wagtail.org/en/latest/contributing/general_guidelines.html#language

File: wagtail/users/tests/test_admin_views.py [hunk 2]
```diff
@@ -1806,3 +1806,257 @@ def test_get_group_viewset_cls_custom_form_does_not_exist(self):
...
+
+    def test_simple(self):
+        response = self.get()
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(
+            response.context["message"],
+            "Sorry, you do not have permission to access this area.",
+        )
+
+    def test_authorized(self):
```
Comment (lb-) [inline wagtail/users/tests/test_admin_views.py:1875]:
```suggestion
    def test_authorised(self):
```

Similar here and the others, using UK spelling.

**Review by lb- — APPROVED** [review]
Looks really great, thank you for being thorough here and writing some solid unit tests.

The tests are good - they do not dive too 'deep' into the output and also check the HTTP status response.

A few minor nitpicks on spelling (I will fix this up and merge).

A general tip for the future, if you include a reference to the Github issue in your commit description it will automatically link the issue to this PR.

```
fixes #9142
```

You can read more about that here.

File: wagtail/users/tests/test_admin_views.py (hunk 1 above)
Comment (mehrdadmoradii) [reply wagtail/users/tests/test_admin_views.py]: Got it! Thanks for letting me know.

**Comment by mehrdadmoradii** [comment]
Thanks for the review @lb-.

**Pull request merged**